pkg/appStore/deployment/AppStoreDeploymentNewService.go

```diff
@@ -0,0 +1,498 @@
+		return false, err
+	}
+	return true, nil
+}
+
+func (impl *AppStoreDeploymentServiceImpl) IsChartRepoActive(appStoreVersionId int) (bool, error) {
+	appStoreAppVersion, err := impl.appStoreApplicationVersionRepository.FindById(appStoreVersionId)
+	if err != nil {
+		impl.logger.Errorw("fetching error", "err", err)
+		return false, err
+	}
+	return appStoreAppVersion.AppStore.ChartRepo.Active, nil
+}
+
+func (impl AppStoreDeploymentServiceImpl) InstallApp(installAppVersionRequest *appStoreBean.InstallAppVersionDTO, ctx context.Context) (*appStoreBean.InstallAppVersionDTO, error) {
+
+	dbConnection := impl.installedAppRepository.GetConnection()
+	tx, err := dbConnection.Begin()
+	if err != nil {
+		return nil, err
+	}
+	// Rollback tx on error.
+	defer tx.Rollback()
+
+	//step 1 db operation initiated
+	installAppVersionRequest, err = impl.AppStoreDeployOperationDB(installAppVersionRequest, tx)
+	if err != nil {
+		impl.logger.Errorw(" error", "err", err)
+		return nil, err
+	}
+
+	if util2.GetDevtronVersion().ServerMode == util2.SERVER_MODE_HYPERION {
+		err = impl.appStoreDeploymentHelmService.InstallApp(installAppVersionRequest, ctx)
+	} else {
+		err = impl.appStoreDeploymentArgoCdService.InstallApp(installAppVersionRequest, ctx)
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	// tx commit here because next operation will be process after this commit.
+	err = tx.Commit()
+	if err != nil {
+		return nil, err
+	}
+
+	//step 4 db operation status update to deploy success
+	_, err = impl.AppStoreDeployOperationStatusUpdate(installAppVersionRequest.InstalledAppId, appStoreBean.DEPLOY_SUCCESS)
+	if err != nil {
+		impl.logger.Errorw(" error", "err", err)
+		return nil, err
+	}
+
+	return installAppVersionRequest, nil
+}
+
+func (impl AppStoreDeploymentServiceImpl) createAppForAppStore(createRequest *bean.CreateAppDTO, tx *pg.Tx) (*bean.CreateAppDTO, error) {
+	app1, err := impl.appRepository.FindActiveByName(createRequest.AppName)
+	if err != nil && err != pg.ErrNoRows {
+		return nil, err
+	}
+	if app1 != nil && app1.Id > 0 {
+		impl.logger.Infow(" app already exists", "name", createRequest.AppName)
+		err = &util.ApiError{
+			Code:            constants.AppAlreadyExists.Code,
+			InternalMessage: "app already exists",
+			UserMessage:     fmt.Sprintf("app already exists with name %s", createRequest.AppName),
+		}
+		return nil, err
+	}
+	pg := &app.App{
+		Active:          true,
+		AppName:         createRequest.AppName,
+		TeamId:          createRequest.TeamId,
+		AppStore:        true,
+		AppOfferingMode: util2.GetDevtronVersion().ServerMode,
+		AuditLog:        sql.AuditLog{UpdatedBy: createRequest.UserId, CreatedBy: createRequest.UserId, UpdatedOn: time.Now(), CreatedOn: time.Now()},
+	}
+	err = impl.appRepository.SaveWithTxn(pg, tx)
+	if err != nil {
+		impl.logger.Errorw("error in saving entity ", "entity", pg)
+		return nil, err
+	}
+
+	// if found more than 1 application, soft delete all except first item
```

Comment:
add sorting to avoid incosistancy

Reply:
Yes but this logic/code is legacy. so avoiding to do as for now